src/pymedphys/msq/_level2/msqdelivery.py

```diff
@@ -149,6 +150,10 @@ def get_mosaiq_delivery_details(cursor, machine, delivery_time, field_label,
     if len(sql_result) > 1:
         for result in sql_result[1::]:
             if result != sql_result[0]:
+                if buffer != 0:
+                    return get_mosaiq_delivery_details(cursor, machine, delivery_time, field_label,
+                                                       field_name, buffer=0)
```

Comment:
Having a buffer can sometimes cause two records to be found that having similar field labels and descriptions. If disagreeing entries are found this is likely due to the buffer. It's hacky to be sure, but here I made it so that if a buffer is defined, and a duplicate entry is found, then in that case try the SQL query again without a buffer.